main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -3064,17 +3064,17 @@ object Weeder2 {
   }
 
   private def visitQName(tree: Tree): Validation[Name.QName, CompilationMessage] = {
-    expect(tree, TreeKind.QName)
     val idents = pickAll(TreeKind.Ident, tree)
     mapN(traverse(idents)(tokenToIdent)) {
-      idents =>
-        assert(idents.nonEmpty) // We require atleast one element to construct a qname
+      case idents if idents.nonEmpty => // We require at least one element to construct a qname
         val first = idents.head
         val ident = idents.last
         val nnameIdents = idents.dropRight(1)
         val loc = SourceLocation(isReal = true, first.loc.sp1, ident.loc.sp2)
         val nname = Name.NName(nnameIdents, loc)
         Name.QName(nname, ident, loc)
+      case nothing => // Return None instead?
+        Name.mkQName("error", SourceLocation.Unknown)
```

Comment:
This is a big problem. Because you are pretending that everything is well and there is a name. Just that it has the text "error". That, unfortunately, won't work.

Reply:
It is not so clear how to fix it though. It may require a Name.Error which would require _a ton_ of refactoring... So let us not do that just yet.

Reply:
What if we return `None` instead in this case and below? Then we do not introduce bad values but we also signal that an error occurred, so we can then collect the error and figure out what to do at the call site

Reply:
Can this actually happen? The old code has some asserts?

Reply:
yes I removed the `expect` call to prevent it from crashing when this case was hit (to test out my implementation)

Reply:
If you return an Option, I suppose you still have to do a lot of refactoring? How many call sites would break?